driver_test.go

```diff
@@ -3209,3 +3209,58 @@ func TestConnectorTimeoutsWatchCancel(t *testing.T) {
 		t.Errorf("connection not closed")
 	}
 }
+
+func TestConnectionAttributes(t *testing.T) {
+	if !available {
+		t.Skipf("MySQL server not running on %s", netAddr)
+	}
+
+	attr1 := "attr1"
+	value1 := "value1"
+	attr2 := "foo"
+	value2 := "boo"
+	dsn += fmt.Sprintf("&connectionAttributes=%s:%s,%s:%s", attr1, value1, attr2, value2)
+
+	var db *sql.DB
+	if _, err := ParseDSN(dsn); err != errInvalidDSNUnsafeCollation {
+		db, err = sql.Open("mysql", dsn)
+		if err != nil {
+			t.Fatalf("error connecting: %s", err.Error())
+		}
+		defer db.Close()
+	}
+
+	dbt := &DBTest{t, db}
+
+	var version string
+	if err := dbt.db.QueryRow("SELECT @@version").Scan(&version); err != nil {
+		dbt.Fatalf("%s", err.Error())
+	}
+	if strings.Contains(strings.ToLower(version), "mariadb") {
+		t.Skip(`TODO: Support adding connection attributes in MariaDB`)
+	}
+
+	var attrValue string
```

Comment:
`ATTR_VALUE` might be `NULL`/`nil`. Maybe not in this specific case. Maybe we should add a `someattr:` to the list?

```suggestion
	var attrValue any
```

Reply:
In the `go.mod` file, the go version is set to 1.13 and it doesn't support `any` .

While `someattr:` will get a "" string value, that is as expected.

Reply:
Using `interface{}` instead of `any` should probably work for older versions of Go. But this is not needed now.